File: process_data.ts
// Random TypeScript function generated on 2025-06-23

interface GetResult {
    id: number;
    name: string;
    value: number;
    active: boolean;
}

function update_value(request_nxd: GetResult[]): GetResult[] {
    const multiplier: number = 2;
    
    return request_nxd
        .filter((item: GetResult) => item.active)
        .map((item: GetResult, index: number) => ({
            ...item,
            value: item.value * multiplier
        }));
}

// Example usage
const testData: GetResult[] = [
    { id: 1, name: "Item1", value: 39, active: true },
    { id: 2, name: "Item2", value: 46, active: false }
];

const result = update_value(testData);
console.log('Processed items:', result);

export { update_value, GetResult };

File: process_data.cpp
// Random C++ function generated on 2025-06-10
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

struct Item {
    int id;
    std::string name;
    double value;
    
    Item(int id, const std::string& name, double value) 
        : id(id), name(name), value(value) {}
};

std::vector<Item> handle_result(const std::vector<Item>& items) {
    const double multiplier = 10.0;
    std::vector<Item> result;
    
    std::transform(items.begin(), items.end(), std::back_inserter(result),
        [multiplier](const Item& item) {
            return Item(item.id, item.name + "_processed", item.value * multiplier);
        });
    
    return result;
}

int main() {
    std::vector<Item> testData = {
        Item(1, "item1", 35.0),
        Item(2, "item2", 82.0)
    };
    
    auto result = handle_result(testData);
    std::cout << "Processed " << result.size() << " items" << std::endl;
    
    return 0;
}
